File: Entrega-2/src/entregaLabo/Estratega.java
package entregaLabo;
import robocode.JuniorRobot;

public class Estratega {
	
	public static StrategySelector estratega1= new Estratega1();
	public static StrategySelector estratega2= new Estratega2();
	
	private static class Estratega1 implements StrategySelector{
		
		@Override
		public RobotActions getStrategy(JuniorRobot robot) {
			if (robot.energy < 30 || robot.others > 5){
				return new ConcreteStrategyInCircles(robot);
			}
			else { 
				return new ConcreteStrategyInline(robot);
			}
		}
		
	}
	
	private static class Estratega2 implements StrategySelector{

		@Override
		public RobotActions getStrategy(JuniorRobot robot) {
			if (robot.others > 2 && robot.energy > 15) {
				return new ConcreteStrategyInCircles(robot);
			}
			else {
				return new ConcreteStrategyInline(robot);
			}
		}
		
	}
}

File: Entrega-4/Estratega.kt
package entregaLabo

import robocode.JuniorRobot

object Estratega {
    val estratega1: StrategySelector = Estratega1()
    val estratega2: StrategySelector = Estratega2()

    private class Estratega1 : StrategySelector {
        override fun getStrategy(robot: JuniorRobot): RobotActions {
            return if (robot.energy < 30 || robot.others > 5) {
                ConcreteStrategyInCircles(robot)
            } else {
                ConcreteStrategyInline(robot)
            }
        }
    }

    private class Estratega2 : StrategySelector {
        override fun getStrategy(robot: JuniorRobot): RobotActions {
            return if (robot.others > 2 && robot.energy > 15) {
                ConcreteStrategyInCircles(robot)
            } else {
                ConcreteStrategyInline(robot)
            }
        }
    }
}
